aea/protocols/generator.py

```diff
@@ -74,6 +75,42 @@
 
 logger = logging.getLogger(__name__)
 
+indent = ""
+
+
+def _change_indent(number: int, mode: str = None) -> None:
+    """
+    Update the value of 'indent' global variable.
+
+    This function controls the indentation of the code produced throughout the generator.
+
+    There are two modes:
+    - Setting the indent to a desired 'number' level. In this case, 'mode' has to be set to "s".
+    - Updating the incrementing/decrementing the indentation level by 'number' amounts. In this case 'mode' is None.
+
+    :param number: the number of indentation levels to set/increment/decrement
+    :param mode: the mode of indentation change
+    :return: None
+    """
+    global indent
```

Comment:
Ugh no global vars please. This will only go wrong! Set a value on the object, for instance the protocol generator ...

Reply:
Done